modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/HorizontalCardTag.java

```diff
@@ -51,185 +48,34 @@ public int doStartTag() throws JspException {
 		return super.doStartTag();
 	}
 
-	public List<DropdownItem> getActionDropdownItems() {
-		if ((_actionDropdownItems == null) && (_horizontalCard != null)) {
-			return _horizontalCard.getActionDropdownItems();
-		}
-
-		return _actionDropdownItems;
+	public HorizontalCard getHorizontalCard() {
+		return (HorizontalCard)getCardModel();
```

Comment:
Is there a reason why we are not doing this like in the old base tag, where we `set` from base
```
public void setHorizontalCard(HorizontalCard horizontalCard) {
	_horizontalCard = horizontalCard;

	super.setBaseClayCard(horizontalCard);
}
```
... and work with private `_horizontalCard` from then on.

To roughly quote the guy from algorithms cource, casting in Java is sus and should be avoided if possible :slightly_smiling_face:

Reply:
No particular reason, I didn't look at the old one. I guess I went for this approach to not store the same object twice (in `BaseCardTag` and on each tag), and casting shouldn't be a problem here since the returned type should be the same as the set one.

Not preference here, this is that kind of java stuff I don't know the expected patterns, pros and cons. Maybe @izaera could help us here, just to know which option is better (if any)

Reply:
Talked with Ivan offline, he agrees on not to create another field to avoid casting, which will penalize runtime, so I'm leaving as it is.